crates/transcode/src/lib.rs

```diff
@@ -277,6 +277,13 @@ impl ContractMessageTranscoder {
             args.push((Value::String(name), value));
         }
 
+        if !data.is_empty() {
+            return Err(anyhow::anyhow!(
```

Comment:
So I don't think we need to bail out entirely here. 

Let's say someone accidentally puts extra bytes into the `decode` command, it could still be useful to show them which bytes were successfully decoded as well as the extra ones.

E.g

```
Managed to decode "get", but had "BE EF" leftover.
```